crates/solver/src/settlement.rs

```diff
@@ -299,6 +303,79 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: u64,
+        tokens_to_satisfy_price_test: &Option<Vec<H160>>,
+    ) -> bool {
+        // The following check is quadratic in run-time, although a similar check with linear run-time would also be possible.
```

Comment:
Since this is `O(n^2)` it might make sense to add an early return at the start of the function if the feature was explicitly disabled (`tokens_to_satisfy_price_test == Some(vec![])`).